src/MoBi.Presentation/Mappers/ModuleConfigurationToModuleConfigurationDTOMapper.cs

```diff
@@ -8,12 +10,32 @@ public interface IModuleConfigurationToModuleConfigurationDTOMapper : IMapper<Mo
    {
 
    }
-   
+
    public class ModuleConfigurationToModuleConfigurationDTOMapper : IModuleConfigurationToModuleConfigurationDTOMapper
    {
+      private readonly IMoBiContext _context;
+
+      public ModuleConfigurationToModuleConfigurationDTOMapper(IMoBiContext context)
+      {
+         _context = context;
+      }
+
       public ModuleConfigurationDTO MapFrom(ModuleConfiguration moduleConfiguration)
       {
-         return new ModuleConfigurationDTO(moduleConfiguration);
+         return new ModuleConfigurationDTO(createProjectModuleConfiguration(moduleConfiguration));
+      }
+
+      private ModuleConfiguration createProjectModuleConfiguration(ModuleConfiguration moduleConfiguration)
+      {
+         // If the module configuration already contains project building blocks then use it
+         var module = _context.CurrentProject.Modules.First(x => Equals(x.Name, moduleConfiguration.Module.Name));
```

Comment:
FindByName
Also using first means it will cras if it's not found. On purpose?

Reply:
Yeah, this should always be able to find the like-named module. Although, at the moment, I'm not sure that if you rename a project module, that it would be renamed in simulation, but that's the idea right?
And you can't remove a used module from the project?